testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -424,6 +443,11 @@ impl BlockMinerThread {
                     "Failed to open chainstate DB. Cannot mine! {e:?}"
                 ))
             })?;
+        if self.last_block_mined.is_some() && self.reason.is_late_block() {
```

Comment:
This makes sense but is not obvious when reading the code. Can you add a comment explaining it? I think something like, "When mining a late block, the miner thread should mine the tenure change block and then exit. Another miner thread will be spawned to continue the tenure in the latest sortition."